src/core/Akka/Actor/ActorCell.DeathWatch.cs

```diff
@@ -87,8 +87,7 @@ public void TerminatedQueuedFor(IActorRef subject)
 
         private bool WatchingContains(IActorRef subject)
         {
-            return _state.ContainsWatching(subject) ||
-                   (subject.Path.Uid != UndefinedUid && _state.ContainsWatching(new UndefinedUidActorRef(subject)));
+            return _state.ContainsWatching(subject);
```

Comment:
`UndefinedUidActorRef` does not need to exist in Akka.NET, because we never had `ActorFor`. All of our actors always have UIDs.